common/rfb/SMsgReader.cxx

```diff
@@ -272,11 +272,32 @@ bool SMsgReader::readKeyEvent()
 
 bool SMsgReader::readPointerEvent()
 {
+  int mask;
+  int x;
+  int y;
+
   if (!is->hasData(1 + 2 + 2))
     return false;
-  int mask = is->readU8();
-  int x = is->readU16();
-  int y = is->readU16();
+
+  mask = is->readU8();
+  x = is->readU16();
+  y = is->readU16();
+
+  if (handler->client.supportsExtendedMouseButtons() && mask & 0x80 ) {
+    int highBits;
+    int lowBits;
+
+    if (!is->hasData(1))
+      return false;
+
+    highBits = is->readU8();
+    lowBits = mask & 0x7f; // Clear marker bit
+    mask = (highBits << 7) | lowBits;
+  } else {
+    // Ignore marker bit
+    mask &= ~(0x7f);
```

Comment:
That is true. We should still read the 8th bit, as we've done previously. This will make our server still support the back button, even without the pseudo-encoding enabled with legacy clients. This is probably how RealVNC works currently.